src/plonk/prover.rs

```diff
@@ -529,50 +529,70 @@ impl<C: CurveAffine> Proof<C> {
                 })
                 .or_else(|| Some(poly));
         }
-        let mut f_poly = f_poly.unwrap();
-        let mut f_blind = Blind(C::Scalar::random());
 
-        let f_commitment = params.commit(&f_poly, f_blind).to_affine();
+        let final_opening;
+        let final_f_commitment;
+        let final_q_evals;
 
-        hash_point(&mut transcript, &f_commitment)?;
+        loop {
+            let mut transcript_dup = transcript.clone();
+            let mut transcript_scalar_dup = transcript_scalar.clone();
+            let mut f_poly_dup = f_poly.clone().unwrap();
 
-        let x_6: C::Scalar = get_challenge_scalar(Challenge(transcript.squeeze().get_lower_128()));
+            let mut f_blind = Blind(C::Scalar::random());
+            let f_commitment = params.commit(&f_poly_dup, f_blind).to_affine();
```

Comment:
We can compute this outside the loop and then within the loop just add `params.h` to it (and increment the corresponding `f_blind`) so that `params.commit` doesn't have to be called more than once.